### PR TITLE
Extensive formatting and code standard changes

### PR DESCRIPTION
**Pull request by richp10**
Please let me know if I have done this correctly / helpfully.  No offense will be taken if you point out I have messed up (still finding my way with git at this level!)

I overwrote some of my earlier changes rather than sort out conflicts - it was quicker to re-run code inspections and re-apply fixes.

### CODE REVIEW

**Comment by richp10** [comment]
Thanks - I will separate out future PR's.

**Comment by amenadiel** [comment]
@richp10 I don't get why now there is a mandatory 4 space indentation of everything, I didn't know that was a standard. If I don't find out how to configure my https://packagecontrol.io/packages/phpfmt to behave in the same way, I'll be stepping over your changes over and over.

Now, there is another issue that haunts me. I've been working in a branch to restore the ability for this software to sub in a subfolder: #40.

When I try to merge with current develop there are at least 10 controllers I can't manually merge. They appear to have changed so much that kdiff3 doesn't even know where to start. Your IDE will probably complain again about code smells on those files, because I'll have to keep them as they are to go on with that feature branch.

**Comment by richp10** [comment]
I had a quick look at phpfmt https://packagecontrol.io/packages/phpfmt - and it *seems* to have the ability to format to PSR standards.  If this continues to cause us problems I *think* I could adjust the settings on my ide to match yours.  On the other hand I could try and persuade you to move to the truly awesome world of jetbrains - you will never go back :)

On second issue - please don't fight with this on my account.  Overwrite my changes and I can always re-run those inspections for the ten controllers. In future I will apply more granular PR's so I hopefully won't cause you this problem in future..

**Comment by amenadiel** [comment]
I've tried PHPStorm but, besides being paid, it feels slow. Perhaps I'll give it another shot as my last experience with it was about two years ago and things might have changed a lot

**Comment by richp10** [comment]
:)  It is pretty resource intensive - I have lots of ram and quite a fast
machine and it has been a revelation; ymmv though..

----------------------------
Richard Phillips
----------------------------

File: src/database/ADODB_base.php [hunk 1]
```diff
@@ -184,10 +184,10 @@ public function fieldClean(&$str)
         /**
          * Cleans (escapes) a string
          *
-         * @param $str The string to clean, by reference
-         * @return The cleaned string
+         * @param  string $str The string to clean, by reference
+         * @return string The cleaned string
          */
-        public function clean(&$str)
+        public function clean(string &$str)
```
Comment (amenadiel) [inline src/database/ADODB_base.php]: @richp10 it seems this type hinting is causing trouble.

See, sometimes `clean` method is used with null parameters, so it's crashing for example when you create a primary key and leave the index name field empty (which is allowed according to Postgres Docs section [sql-altertable.html](https://www.postgresql.org/docs/9.6/static/sql-altertable.html))

I believe this is the only scalar typehinting you added, so I removed it in the last tag

**Pull request merged**